staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -240,7 +240,8 @@ func (s *DelegatingAuthenticationOptions) getClientCA() (*ClientCertAuthenticati
 		return nil, err
 	}
 	if incluster == nil {
-		return nil, fmt.Errorf("cluster doesn't provide client-ca-file")
+                glog.Warningf("cluster doesn't provide client-ca-file")
```

Comment:
probably worth making this warning more explicit about what it was looking for, and what usually should be done in the main apiserver to populate that configmap. also worth warning that this is continuing without the ability to use x509-based authentication

Reply:
it seems like `getRequestHeader()` should be consistent with this as well... if we successfully look up the options from the configmap, and get no info, should `getRequestHeader()` log a warning that we're continuing without request header auth, and continue?

Reply:
@liggitt I didn't make the same change for request header because as per my understanding delegated authentication requires the requestheader client CA to trust the aggregator, hence failing when it isn't present is the right behavior. @DirectXMan12 What do you think?

Reply:
Yeah, I think there's a lot more of an argument for "I want requestheader auth without normal client cert auth" than vice-versa (I don't see the case for auto-auth setup with *only* normal client cert auth).

Do agree with the warning being more explicit -- one of the reasons I actually suggested an option, so that you have to manually opt in to this behavior if you want it.

Reply:
I'm open to making the log message more explicit as suggested though unsure why a new flag is necessary to opt in to building the extension API server without client-ca? What's the risk in that case?

Reply:
> I think there's a lot more of an argument for "I want requestheader auth without normal client cert auth" than vice-versa

That's probably true. Forget I said anything.